x-pack/metricbeat/module/prometheus/remote_write/data.go

```diff
@@ -182,12 +186,31 @@ func (g remoteWriteTypedGenerator) GenerateEvents(metrics model.Samples) map[str
 			histograms[histKey] = hist
 			continue
 		}
-		e.ModuleFields.Update(data)
 
+		e.ModuleFields.Update(data)
 	}
 
 	// process histograms together
 	g.processPromHistograms(eventList, histograms)
+
+	if g.metricsCount {
+		for _, e := range eventList {
+			// In x-pack prometheus module, the metrics are nested under the "prometheus" key directly.
+			// whereas in non-x-pack prometheus module, the metrics are nested under the "prometheus.metrics" key.
+			// Also, it is important that we do not just increment by 1 for each e.ModuleFields["metrics"] may have more than 1 metric.
+			// As, metrics are nested under the "prometheus" key, labels is also nested under the "prometheus" key. So, we need to make sure
+			// we subtract 1 in case the e.ModuleFields["labels"] also exists.
+			//
+			// See unit tests for the same.
+			_, ok := e.ModuleFields["labels"]
+			if ok {
+				e.RootFields["metrics_count"] = len(e.ModuleFields) - 1
+			} else {
+				e.RootFields["metrics_count"] = len(e.ModuleFields)
+			}
```

Comment:
I think it will be more intuitive to name the flags that actually denotes the meaning of the flag. 
```suggestion
			if _, hasLabels := e.ModuleFields["labels"]; hasLabels {
				e.RootFields["metrics_count"] = len(e.ModuleFields) - 1
				continue
			} 	
			e.RootFields["metrics_count"] = len(e.ModuleFields)
```

Reply:
Cool. s/ok/hasLabels in related pieces of code.

Also for consistency across the setting of metrics_count related changes in collector and remote_write, using if-else (not if along with continue) pattern because here it works fine but a similar pattern with `continue` in collector won't work because the event is being reported in the very same loop. So, keeping it if-else pattern; this is also idiomatic and it's pretty clear too.

Reply:
Changes: https://github.com/elastic/beats/pull/40411/commits/b1408a73521d639ca80f90b7362b2a63e778829f